lib/titiler-pgstac-api/index.ts

```diff
@@ -40,15 +40,15 @@ import {
       this.titilerPgstacLambdaFunction = new lambda.Function(this, "lambda", {
         handler: "handler.handler",
         runtime: lambda.Runtime.PYTHON_3_10,
-        code: lambda.Code.fromDockerBuild(__dirname, {
+        code: props.titilerApiAsset ?? lambda.Code.fromDockerBuild(__dirname, {
           file: "runtime/Dockerfile",
           buildArgs: { PYTHON_VERSION: '3.10' },
         }),
         timeout: Duration.seconds(30),
         vpc: props.vpc,
         vpcSubnets: props.subnetSelection,
         allowPublicSubnet: true,
-        memorySize: 3008,
+        memorySize: props.titilerLambdaMemorySize ?? 3008,
         logRetention: aws_logs.RetentionDays.ONE_WEEK,
         environment: titilerPgstacEnv,
       });
```

Comment:
Can we switch to `PythonFunction` like for the other construct?

Reply:
Wasn't the purpose of using a docker build to be able to [trim a little bit the package through the dockerfile ?](https://github.com/developmentseed/eoapi-cdk/blob/a94cf94cd8c0d7270cacb678f545a7bc14a54af9/lib/titiler-pgstac-api/runtime/Dockerfile#L12) 

If we move to `PythonFunction`, [can we do the same for the database `bootstrapper`? ](https://github.com/developmentseed/eoapi-cdk/blob/a94cf94cd8c0d7270cacb678f545a7bc14a54af9/lib/bootstrapper/index.ts#L32)

Reply:
> As is, it doesn't work with PythonFunction because the package is too big.

Looks like `boto3` (which requires `botocore`) is the main cuplrit, it's taking a third of the size. It looks like it's supposed to be included in the lambda python runtime, so we shouldn't need to install it. I am going to remove it from the main requirements to the dev requirements and see if I can deploy this as a `PythonFunction` with that trick.

Reply:
Ok that works in my last commit

Reply:
Looking at the CdK docs I though we could still use custom docker build ... but maybe not! That will be definitely a blocker if not